sky/packages/sky/lib/src/fn3/basic.dart

```diff
@@ -44,13 +83,653 @@ class DecoratedBox extends OneChildRenderObjectWidget {
+    renderObject.width = width;
+    renderObject.height = height;
+    renderObject.colorFilter = colorFilter;
+    renderObject.fit = fit;
+    renderObject.repeat = repeat;
+  }
+}
+
+class ImageListener extends StatefulComponent {
+  ImageListener({
+    Key key,
+    this.image,
+    this.width,
+    this.height,
+    this.colorFilter,
+    this.fit: ImageFit.scaleDown,
+    this.repeat: ImageRepeat.noRepeat
+  }) : super(key: key) {
+    assert(image != null);
+  }
+
+  final ImageResource image;
+  final double width;
+  final double height;
+  final sky.ColorFilter colorFilter;
+  final ImageFit fit;
+  final ImageRepeat repeat;
+
+  ImageListenerState createState() => new ImageListenerState(this);
+}
+
+class ImageListenerState extends ComponentState<ImageListener> {
+  ImageListenerState(ImageListener config) : super(config) {
+    config.image.addListener(_handleImageChanged);
+  }
+
+  sky.Image _resolvedImage;
+
+  void _handleImageChanged(sky.Image resolvedImage) {
+    setState(() {
+      _resolvedImage = resolvedImage;
+    });
+  }
+
+  void dispose() {
+    config.image.removeListener(_handleImageChanged);
+  }
+
+  void didUpdateConfig(ImageListener oldConfig) {
+    if (config.image != oldConfig.image) {
+      oldConfig.image.removeListener(_handleImageChanged);
+      config.image.addListener(_handleImageChanged);
+    }
+  }
+
+  Widget build() {
+    return new Image(
+      image: _resolvedImage,
+      width: config.width,
+      height: config.height,
+      colorFilter: config.colorFilter,
+      fit: config.fit,
+      repeat: config.repeat
+    );
+  }
+}
+
+class NetworkImage extends StatelessComponent {
+  NetworkImage({
+    Key key,
+    this.src,
+    this.width,
+    this.height,
+    this.colorFilter,
+    this.fit: ImageFit.scaleDown,
+    this.repeat: ImageRepeat.noRepeat
+  }) : super(key: key);
+
+  final String src;
+  final double width;
+  final double height;
+  final sky.ColorFilter colorFilter;
+  final ImageFit fit;
+  final ImageRepeat repeat;
+
+  Widget build() {
+    return new ImageListener(
+      image: imageCache.load(src),
+      width: width,
+      height: height,
+      colorFilter: colorFilter,
+      fit: fit,
+      repeat: repeat
+    );
+  }
+}
+
+class AssetImage extends StatelessComponent {
+  AssetImage({
+    Key key,
+    this.name,
+    this.bundle,
+    this.width,
+    this.height,
+    this.colorFilter,
+    this.fit: ImageFit.scaleDown,
+    this.repeat: ImageRepeat.noRepeat
+  }) : super(key: key);
+
+  final String name;
+  final AssetBundle bundle;
+  final double width;
+  final double height;
+  final sky.ColorFilter colorFilter;
+  final ImageFit fit;
+  final ImageRepeat repeat;
+
+  Widget build() {
+    return new ImageListener(
+      image: bundle.loadImage(name),
+      width: width,
+      height: height,
+      colorFilter: colorFilter,
+      fit: fit,
+      repeat: repeat
+    );
+  }
+}
+
+// TODO(abarth): WidgetToRenderBoxAdapter
```

Comment:
heh, I was wondering how you'd do this one. I think we should do this one sooner rather than later since it affects the design of RenderObjectElement.

Maybe it should just have a wrapper RenderObject and manipulate that, rather than actually changing its own renderObject, though.